pkg/queue/bounded_queue.go

```diff
@@ -39,9 +37,9 @@ type Consumer interface {
 type BoundedQueue struct {
 	workers       int
 	stopWG        sync.WaitGroup
-	size          *uatomic.Uint32
-	capacity      *uatomic.Uint32
-	stopped       *uatomic.Uint32
+	size          atomic.Int32
```

Comment:
atomic.Uint32 does not support subtracting, so switch to signed. OTEL collector has a cleaner implementation of bounded queue that we will be able to use in jaeger-v2